### DISCUSSION TITLE
Improve code maintainability

### DISCUSSION THREAD

**Discussion by sdondley** [jbyuki/instant.nvim#30] [General]
I just made a couple of pull requests. During this process, I found the code very hard to follow. Specifically, the StartClient function has ~2000 lines of code packed with dozens of if/else statements that could run several pages long. Perhaps there is a good reason for this but if there isn't I'd be interested into doing what I can to make the code more readable and maintainable. It would be a good way for me to get more familiar with lua programming. Let me know. Thanks.

**Comment by jbyuki** [comment]
I must say I have not been very active on this repository. If you can help make the code more readable and maintenable, I can grant you access as a collaborator to this repo. I have sent you the request. I wish to make this project more community driven.

**Reply by sdondley** [reply]
Ok cool. I'll work on the basic documentation first so I can get more familiar with the plug-in. I need to get more familiar with Lua before I make any major changes to the code but I'll work on smaller improvements until then to also help me figure out my way around. Can you spell out how to test lua code with neovim or point me to a little tutorial?

**Reply by jbyuki** [reply]
I'm sure it will take some time until someone can get familiar with it. Lots of trial and error. But a good starting point is [this](https://github.com/nanotee/nvim-lua-guide) if you havent seen it yet.
